### PR TITLE
Misc admin toys and WiP stuff

### PR DESCRIPTION
**Pull request by Tarmunora**
[FEATURE]
## About the PR 
Adds fullauto and smartgun components, and preset guns to use them
Also adds an 'autoaim' pistol that automatically shoots at the closest mob nearby where you click
Additionally adds scrolls of enchantment and associated properties/handling, and an `enchant()` proc to add/remove enchantment directly

Nothing player-facing currently (will probably make the +2 robes of dungeon mastery use the enchantment system in the future, and trial the gun stuff for nukeops)
## Why's this needed? 
Cool toys, fun to play with

### CODE REVIEW

File: _std/defines/component_defines.dm [hunk 1]
```diff
@@ -144,3 +149,10 @@
 // obj/critter signals
 // When an obj/critter dies
 #define COMSIG_OBJ_CRITTER_DEATH "obj_critter_death"
+
+/// general return values //
+#define RETURN_SUCCESS 0x001
+#define RETURN_FAILURE 0x002
```
Comment (pali6) [inline _std/defines/component_defines.dm:155]: Are these supposed to be general purpose flags for returning from signals? If so I don't think that's a good idea. I'd rather have each signal define its own, especially considering how these likely clash with already existing return flags. (Also it's a bit weird how you can have success and failure at the same time. I'd expect success to be one flag and the absence of success deemed as a failure.)

Comment (Tarmunora) [reply _std/defines/component_defines.dm]: on second though I wasn't even using those anyways

File: code/modules/interface/mob_input.dm [hunk 2]
```diff
@@ -83,7 +83,8 @@
 		if (abilityHolder.topBarRendered)
 			if (abilityHolder.click(target, params))
 				return 100
-	SEND_SIGNAL(src, COMSIG_MOB_CLICK, target, params)
+	if(SEND_SIGNAL(src, COMSIG_MOB_CLICK, target, params) && RETURN_CANCEL_CLICK)
+		return 100
```
Comment (pali6) [inline code/modules/interface/mob_input.dm:87]:
```suggestion
	if(SEND_SIGNAL(src, COMSIG_MOB_CLICK, target, params) & RETURN_CANCEL_CLICK)
		return 100
```
Did you even test that?

**Review by pali6 — CHANGES_REQUESTED** [review]

**Review by pali6 — APPROVED** [review]

File: code/WorkInProgress/ObjectProperties.dm [hunk 3]
```diff
@@ -101,6 +101,8 @@ var/list/globalPropList = null
 	var/tooltipImg = "" //Stored in browserassets\images\tooltips
 	var/defaultValue = 1 //Default value. Used to get an idea of what's "normal" for any given property.
 	var/goodDirection = 1 //Dumb name. Tells us which direction the number should grow in for it to be considered "good", 1=positive, -1 negative
+	var/hidden = 0
+	var/inline = 0
```
Comment (ZeWaka) [inline code/WorkInProgress/ObjectProperties.dm:105]: can we get some documentation for these

Comment (Tarmunora) [reply code/WorkInProgress/ObjectProperties.dm]: yes

**Pull request merged**